internal/kademlia/contact/contact.go

```diff
@@ -23,27 +23,24 @@ func NewContact(id kademliaid.KademliaID, address string) *Contact {
 
 // CalcDistance calculates the distance to the target and
 // fills the contacts distance field
-// TODO: target should be a contact
 func (contact *Contact) CalcDistance(target kademliaid.KademliaID) {
 	contact.distance = contact.ID.CalcDistance(target)
 }
 
 // Less returns true if contact.distance < otherContact.distance
-// TODO: Don't know if we use it in this way, lets see
 // We just compare distances here
-// TODO: It implements the comparable interface or smth like this, for sorting
 func (contact *Contact) Less(otherContact *Contact) bool {
 	return contact.distance.Less(otherContact.distance)
 }
 
 // String returns a simple string representation of a Contact
-// TODO: It should implement the standard toString interface, check this
 func (contact *Contact) String() string {
 	return fmt.Sprintf(`contact("%s", "%s")`, contact.ID, contact.Address)
 }
 
-func SortContacts(contacts *[]Contact) {
-	slices.SortStableFunc(*contacts, func(a, b Contact) int {
+// SortContacts sorts a slice of contacts
+func SortContacts(contacts []Contact) {
```

Comment:
why do we want a copy of the slice here? It should break this code, doesn't it? Because a copy is passed and then the copy is sorted. So nothing has changed for the caller

Reply:
This should fail in a test, doesn't it?